### ISSUE TITLE
Make targets widgets clickable and support target detail pages for supervisors

### ISSUE THREAD

**Issue by amandacilek** [medic/cht-core#4840]
[Product initiative detail](https://docs.google.com/document/d/1EQHQI11XqI8aQeUsnmlrD3ny83fj5N6MZATrLzsm_-I/edit#heading=h.q31g948sivit)

Read more details and view additional screenshots in the MVP deck: https://docs.google.com/presentation/d/1AbOxkGgggUoI0QzRmwVqfX0AqC6aG6CByNd-CzdRgaI/edit?usp=sharing

This new feature builds off of changes made in ticket #4839. The detail pages are considered a nice-to-have but not essential for the Supervisor MVP. Depends on further technical investigation and level of effort.

**New feature: targets detail pages**
- Aggregate widgets are now clickable
- Clicking on a widget takes you to a new detail page view of that widget with a breakdown of data by individual CHW
- The title of the detail page should match the title of the widget
- CHW names on the left are hyperlinked to a CHW’s profile
- The individual performance of each CHW is displayed on the right (could be a count or a percentage)
- The individual performance values are right-aligned 
- If the data is a percentage, we should also display the counts in parenthesis e.g. “90% (9 of 10)” 
- The list should be automatically sorted from lowest to highest performing and/or lowest to highest value. Lowest performing at the top.
- If a particular widget has a goal, it would be nice to support conditional styling based upon goals: If a certain CHW is performing below the goal, their value should be highlighted in red. If a CHW is performing above the goal, their value should be green.

Note from @SCdF : I think this is just visual: we should be able to make a new URL target that has the target in it and then displays all families by target, and ideally ordering is guessable based on the target type.

![screen shot 2018-09-19 at 1 01 11 pm](https://user-images.githubusercontent.com/26235957/45778016-40f18680-bc0c-11e8-9b18-aa6485f0c459.png)

**Comment by MaxDiz** [comment]
@wtekeu please hold off on development for this ticket and #4839. We have a Community Partner co-development opportunity and are working on a more comprehensive plan for Supervisor app buildout

**Comment by n-orlowski** [comment]
Hi all, please see further designs for tablet and desktop styling. Please ping me if anything is unclear. 

<img width="1097" alt="Target Detail" src="https://user-images.githubusercontent.com/59581790/72467968-14229480-37aa-11ea-8e24-2bac43eb5c69.png">

Sketch file: https://drive.google.com/open?id=1-2r2Ithqtez2yLn1LIJQFs_J24bzCnj-

**Comment by n-orlowski** [comment]
Three things to look at:

- Two target views (Supervisor's own vs aggregate CHWs) - how does this work? Dropdown? Toggle?
- Supervisor tab: Existing widgets
- CHW tab: LHS panel, RHS detailed content

Will come back with potential designs

**Comment by n-orlowski** [comment]
<img width="1467" alt="Desktop" src="https://user-images.githubusercontent.com/59581790/72936947-13937c00-3d1d-11ea-9036-3e109f84d989.png">
<img width="672" alt="Mobile" src="https://user-images.githubusercontent.com/59581790/72936948-142c1280-3d1d-11ea-862b-f9a02edda7fb.png">
<img width="527" alt="Tablet" src="https://user-images.githubusercontent.com/59581790/72936949-142c1280-3d1d-11ea-844a-71640e4f02d8.png">

@dianabarsan here are the designs we talked about. Let me know if there are any questions!

**Comment by n-orlowski** [comment]
<img width="1461" alt="Desktop" src="https://user-images.githubusercontent.com/59581790/73089434-190ed480-3e8b-11ea-8161-07fcde414a47.png">
<img width="632" alt="Mobile" src="https://user-images.githubusercontent.com/59581790/73089435-190ed480-3e8b-11ea-9601-5874b7fb641f.png">
<img width="961" alt="Tablet" src="https://user-images.githubusercontent.com/59581790/73089436-190ed480-3e8b-11ea-97b2-cffd4ca2edce.png">

@dianabarsan Please see revised screens: 
- Empty states have 3 instances as outlined in desktop (although not sure which are possible at this time). 
- Line items are sorted alphabetically for MVP
- Date stamp is removed from sidebar and title and is moved to the RHS of the content panel (in later iterations we may be able to view past 3 months, 6 months, all time etc). 
- Label "Reports" is not the right term but a placeholder for now (still waiting on feedback from the team on this one). 
- All views minus count with no goal are displayed with progress bars

**Comment by dianabarsan** [comment]
@n-orlowski 
In regards to the date stamp (which appears as "This month" in your mockups): in the original Targets mockups and the current implementation, this is, in fact, the target's configured subtitle. 

In your "My targets" screen, you have 2 variants of this subtitle: "This month" and "All-time", but these can be literally any text the configurer chooses.

Also, targets can support configurations that count all-time entries (hence the "All-time" Target example in the mockups - and our default configuration even has a couple of these). 
This means that the values the are stored in the targets represent the total of all-time entries, so "This month" doesn't cut it in this case, and using the Target's subtitle would be a the better choice. 

Should I use the Target's subtitle for the "Date stamp" (like the Targets page already does)?


In regards to the "empty states", Targets don't currently support setting different goals per user, so the goal is shared among all and is defined in the target configuration, which we always have access to. So, it's not possible to have an "unknown goal". Are you ok with scrapping the "goal unknown" state?

**Comment by dianabarsan** [comment]
I'm also noticing that the color is for "goal not met" is changed compared to the color we use now: ##E33030 (old) vs #e55309 (new):
![image](https://user-images.githubusercontent.com/35681649/73180038-28a54c00-411d-11ea-978c-bef4ef7b1c6c.png)

Is this intended or a miss? 
The black text (as indicated in the mockups) doesn't look very good in over the old color, but white does look decent:

![image](https://user-images.githubusercontent.com/35681649/73180196-820d7b00-411d-11ea-912e-eab797ddd6a2.png)

![image](https://user-images.githubusercontent.com/35681649/73180231-9487b480-411d-11ea-8031-470bac03677b.png)

**Comment by n-orlowski** [comment]
@dianabarsan The reason I updated it to not have the subtitles "this month" or "all-time" is because this feature aligns with the new DHIS2 requests where users want to see trends in targets/performance (ie to easily compare between months, similar to Reports on profile pages). If it's a huge lift to implement it like this I'm happy to use what we have already configured for MVP to test and then iterate in later versions. What are your thoughts?

As for colours, I had used what I found in an existing Sketch file but that was a miss! Let's use the new colour. I opted for black on the bars since it's the one colour that is accessible (colour-contrast-wise) against all variations of the bar (teal, grey empty state and orange):

https://webaim.org/resources/contrastchecker/?fcolor=333333&bcolor=7AB1B1
https://webaim.org/resources/contrastchecker/?fcolor=333333&bcolor=E55309

I agree white looks better but black functions better across the range of colours we have. 

Also happy to scrap the goal unknown! This was brought to my attention as a possible requirement but if it's not necessary let's not bother with it.

cc @maremegaye

**Comment by dianabarsan** [comment]
@n-orlowski 
In terms of the subtitle, I was referring to the text in the header ("Reports" and "This month"). 
Would you like to display "This month" across all target types, even the ones that count all-time values, or use the target's subtitle there instead? 

In terms of colors and in the interest of keeping consistency, I will update the color we use in the "Targets" page as well.

**Comment by dianabarsan** [comment]
Ready for AT on `4839-aggregate-targets` - along with #4839

- should list contacts in alphabetical order
- clicking on a line contact should redirect you to the contact detail page.

**Comment by ngaruko** [comment]
Tested along with #4839 - LGTM @dianabarsan .
Please feel free to merge and close.

**Comment by dianabarsan** [comment]
Merged.
